eind-opdracht/be/src/main/java/kbs/controller/CompanyController.java

```diff
@@ -28,21 +33,21 @@ public ResponseEntity<CompanyDto> getCompany(@PathVariable Long id) {
     }
 
     @PostMapping
-    public ResponseEntity<Object> createCompany(@RequestBody CompanyDto cdto, BindingResult bindingResult) {
-
+    public ResponseEntity<Object> createCompany(@Valid @RequestBody CompanyDto cdto, BindingResult bindingResult) {
         if (bindingResult.hasFieldErrors()) {
-            StringBuilder sb = new StringBuilder();
-            for (FieldError fe : bindingResult.getFieldErrors()) {
-                sb.append(fe.getField()).append(": ");
-                sb.append(fe.getDefaultMessage()).append("\n");
-            }
-            return new ResponseEntity<>(sb.toString(), HttpStatus.BAD_REQUEST);
+            GsonBuilder builder = new GsonBuilder();
+            builder.registerTypeHierarchyAdapter(BindingResult.class, new BindingResultFieldErrorAdapter());
+            builder.setPrettyPrinting();
+            Gson gson = builder.create();
+            return new ResponseEntity<>(gson.toJson(bindingResult), HttpStatus.BAD_REQUEST);
```

Comment:
Returns `bindingResult.fieldErrors` in JSON format to be used FE

e.g. 
<img width="830" alt="image" src="https://github.com/manonlammers/novi/assets/13642402/0fe221d3-8c9a-4c26-bb17-e12ff59df33d">